kuka_mobile_robot_support/ros2_control/mrob_kuka.ros2_control.xacro

```diff
@@ -0,0 +1,96 @@
+<?xml version="1.0"?>
+<robot xmlns:xacro="http://www.ros.org/wiki/xacro">
+
+  <xacro:macro name="mrob_kuka_ros2_control" params="name prefix client_ip controller_ip use_fake_hardware qos_config_file roundtrip_time">
+
+    <ros2_control name="${name}" type="system">
+      <hardware>
+        <xacro:if value="${use_fake_hardware}">
+          <plugin>kuka_mock_hardware_interface::KukaMockHardwareInterface</plugin>
+          <param name="cycle_time_ms">4</param>
+          <param name="roundtrip_time_micro">${roundtrip_time}</param>
+          <param name="mock_sensor_commands">false</param>
+          <param name="state_following_offset">0.0</param>
+          
+          <!--<xacro:if value="${gazebo_sim}">
+            <plugin>kuka_eac::KukaEACHardwareInterface</plugin>
+            <plugin>gazebo_ros2_control/GazeboSystem</plugin>
+          </xacro:if> -->
+        </xacro:if>
+        <xacro:unless value="${use_fake_hardware}">
+          
+          <!-- Read QoS profile parameters  -->
+          <xacro:if value="${qos_config_file == ''}">
+            <xacro:property name="qos_config" value="$(find kuka_iiqka_eac_driver)/config/qos_config.yaml"/>
+          </xacro:if>
+          <xacro:unless value="${qos_config_file == ''}">
+            <xacro:property name="qos_config" value="${qos_config_file}"/>
+          </xacro:unless>
+          <xacro:property name="qos_config_dict" value="${xacro.load_yaml(qos_config)}"/>
+          <xacro:property name="consequent_lost_packets" value="${qos_config_dict['rt_packet_loss_profile']['consequent_lost_packets']}"/>
+          <xacro:property name="lost_packets_in_timeframe" value="${qos_config_dict['rt_packet_loss_profile']['lost_packets_in_timeframe']}"/>
+          <xacro:property name="timeframe_ms" value="${qos_config_dict['rt_packet_loss_profile']['timeframe_ms']}"/>
+          <plugin>kuka_eac::KukaEACHardwareInterface</plugin>
+          <param name="client_ip">${client_ip}</param>
+          <param name="controller_ip">${controller_ip}</param>
+          <param name="consequent_lost_packets">${consequent_lost_packets}</param>
+          <param name="lost_packets_in_timeframe">${lost_packets_in_timeframe}</param>
+          <param name="timeframe_ms">${timeframe_ms}</param>
+        </xacro:unless>
+      </hardware>
+      <!-- define joints and command/state interfaces for each joint -->
+      <joint name="${prefix}twist/linear">
```

Comment:
if the state/command interfaces are not connected to an actual joint, it is not necessary to give them in the urdf/xacro, you can simply export the interfaces in the hwif and you can access those in the controller

the joints you defined here are not actual joints, rather a workaround, so I would remove these definitions

Reply:
resolved only due to follow-up comment in new file